### PR TITLE
Initial work for unit tests and docs

### PR DESCRIPTION
**Pull request by Kicer86**

### CODE REVIEW

File: Doxyfile [hunk 1]
```diff
@@ -0,0 +1,2538 @@
...
+
+# If the GENERATE_ECLIPSEHELP tag is set to YES, additional index files will be
+# generated, together with the HTML files, they form an Eclipse help plugin. To
+# install this plugin and make it available under the help contents menu in
+# Eclipse, the contents of the directory containing the HTML and XML files needs
+# to be copied into the plugins directory of eclipse. The name of the directory
+# within the plugins directory should be the same as the ECLIPSE_DOC_ID value.
+# After copying Eclipse needs to be restarted before the help appears.
+# The default value is: NO.
+# This tag requires that the tag GENERATE_HTML is set to YES.
```
Comment (ThudPoland) [inline Doxyfile]: This file shouldn't be part of project. It should be downloaded by CMake, beacuse it's not a part of code, just definitions for generating doxygen.

File: basic/IObject.hpp [hunk 2]
```diff
@@ -4,20 +4,36 @@
...
+ *  It is a part of game model. It does not contain any
+ *  graphical information.
+ */
+
 class IObject
 {
 public:
-    virtual const sf::Vector2f& getPosition() const = 0;
-    virtual const float getRotationDegrees() const = 0;
+    virtual const sf::Vector2f& getPosition() const = 0;        ///< Returns object position. \return object position.
```
Comment (ThudPoland) [inline basic/IObject.hpp]: Comments for methods should be in source file.
Making such long lines make code unreadable.

File: player/PlayerCreator.hpp [hunk 3]
```diff
@@ -5,6 +5,10 @@
 #include "GameManager.hpp"
 #include "Player.hpp"
 
+/**
+ * \brief Builder for \ref Player object.
```
Comment (ThudPoland) [inline player/PlayerCreator.hpp]: It's rather creator, it don't work now as builder.

File: basic/IObject.hpp (hunk 2 above)
Comment (Kicer86) [reply basic/IObject.hpp]: yet it is easier to find methods description in hpp file

File: Doxyfile (hunk 1 above)
Comment (Kicer86) [reply Doxyfile]: Where would you like to download it from?

Comment (Kicer86) [reply Doxyfile]: where would you like to download it from?

**Pull request merged**